### PR TITLE
Eliminates padding from some common structs

### PR DESCRIPTION
**Pull request by zwoop**
There's still plenty to be done here, but this is pretty tedious work ... And have to be careful too, which I'm not known for ...

### CODE REVIEW

**Comment by masaori335** [comment]
+1. Great! I've wanted this for a long time! Let's tidy up little by little.

**Comment by zwoop** [comment]
Yeh, we've done this in the past as well, things just creeps up. I think longer term (intern maybe, good idea Craig) would be to make a very consistent layout of all structs and classes (in this order), one section for public and one section for private:

1. All 64-bit types / pointers // (these will never be padded, so could go last as well
2. All enum types // These can vary in size, I think, typically 2 bytes maybe ?
3. All bitfields
4. All bool types, or other 1 byte types
5. All 32-bit types

This won't solve all padding issues, but, it at least avoids a lot of them. Our HttpSM.h is particularly messy, I fixed some of it, but it was going back and forth between public / private. I fixed some of that, but not all.

Additionally, it can help to arrange  the public vs private sections in a way that the padding between the two is minimized. This may require reordering the list above around those boundaries.

I used a tool call pahole for this. It's definitely not great, because of our class structures and stuff, and doesn't do an awesome job with classes. Maybe there are better tools to investigate, or, maybe just use `-Wpadding` ?

File: iocore/eventsystem/I_EventProcessor.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #ifdef TS_MAX_THREADS_IN_EACH_THREAD_TYPE
 constexpr int MAX_THREADS_IN_EACH_TYPE = TS_MAX_THREADS_IN_EACH_THREAD_TYPE;
 #else
-constexpr int MAX_THREADS_IN_EACH_TYPE = 3072;
+constexpr int MAX_THREADS_IN_EACH_TYPE = 3071;
```
[the remark this answers is not in the document]
Comment (zwoop) [reply iocore/eventsystem/I_EventProcessor.h:35]: It was pretty random, I was going to restore it but ran into a squirrel, but making that odd also made it 16 bytes smaller for some reason ... Should we restore it? It'd be surprising if 3072 is a magic number that actually matters?

[the remark this answers is not in the document]
Comment (ywkaras) [reply iocore/eventsystem/I_EventProcessor.h:35]: Nah ya never know when you'll need 16 bytes.

**Pull request merged**